clients/privacy-center/pages/consent.tsx

```diff
@@ -214,16 +221,61 @@ const Consent: NextPage = () => {
     redirectToIndex,
   ]);
 
+  /**
+   * Initialize internalization library.
+   */
+  const experience = useAppSelector(selectPrivacyExperience);
+  const { IS_OVERLAY_ENABLED } = settings;
+  const isConfigDrivenConsent = !IS_OVERLAY_ENABLED;
+  const [isI18Initialized, setIsI18Initialized] = useState(false);
+
+  useEffect(() => {
+    const i18n = setupI18n();
+
+    // 1. Config driven consent
+    if (isConfigDrivenConsent) {
+      // If we're in config driven consent then
+      // still load the static messages in order to have default english
+      // messages available
+      loadMessagesFromFiles(i18n);
+      setI18nInstance(i18n);
+      setIsI18Initialized(true);
+      return;
+    }
+
+    // 2. Notice driven consent
+    if (!experience) {
+      return;
+    }
+
+    initializeI18n(
+      i18n,
+      window?.navigator,
+      experience as PrivacyExperience,
+      {
+        debug: true,
+      },
+      {}
+    );
+
+    setI18nInstance(i18n);
+    setIsI18Initialized(true);
+  }, [experience, setI18nInstance, isConfigDrivenConsent]);
+
   return (
     <Stack as="main" align="center" data-testid="consent">
-      <Stack align="center" py={["6", "16"]} spacing={8} maxWidth="720px">
-        <Stack align="center" spacing={3}>
-          <ConsentHeading />
-          <ConsentDescription />
+      {/* Wait until i18n is initalized so we can diplay the correct language and
+       also we can use the correct history ids */}
+      {isI18Initialized && (
```

Comment:
We try to use `condition ? <Component /> : null` rather than `condition && <Component/>` for JSX to prevent confusing behavior with falsy-but-not-false values-- reasoning is explained [here](https://kentcdodds.com/blog/use-ternaries-rather-than-and-and-in-jsx).  It's not totally applicable in this case because `isI18nInitialized` should never be anything other than exactly `true` or `false` but it's easier to just always do it this way by convention.

Reply:
Didn't know that. Just applied the change. thanks.

Reply:
cc @gilluminate

Reply:
@jpople that's easily solved by always forcing boolean on non-boolean types.
for example, in the article you referenced here:
```
!!contacts.length && ...
```
Personally I find it to be a much cleaner approach and the way that React documentation suggests it should be written.